app/src/main/java/nu/yona/app/api/manager/impl/NotificationManagerImpl.java

```diff
@@ -161,6 +162,87 @@ public void onError(Object errorMessage) {
         }
     }
 
+
+    public void getMessageWithUrl(final String urlForMessagesFetch, final boolean isUnreadStatus, final DataLoadListener listener, final boolean isProcessUpdate) {
+        try {
+            User user = YonaApplication.getEventChangeManager().getDataState().getUser();
+            if (user != null && user.getLinks() != null && user.getLinks().getYonaMessages() != null
+                    && !TextUtils.isEmpty(user.getLinks().getYonaMessages().getHref())) {
+
+                notificationNetwork.getNextSetOfMessagesFromURL(urlForMessagesFetch, YonaApplication.getEventChangeManager().getSharedPreference().getYonaPassword(), isUnreadStatus, new DataLoadListener() {
+                    @Override
+                    public void onDataLoad(Object result) {
+                        if (listener != null) {
+                            if (result instanceof YonaMessages) {
```

Comment:
Can you realistically be called with other results than Yona messages?

Reply:
I don't know why they had this check. 
But I faced some issues with ErrorMessage class. Some times (When ) the call back comes with all properties to cast the object to ErrorMessage and some times it will only come with string message="some message" and will not have error code. So it better to have the check.

Reply:
OK